src/nb_enum.cpp

```diff
@@ -113,6 +119,23 @@ void enum_append(PyObject *tp_, const char *name_, int64_t value_,
         fail("refusing to add duplicate key \"%s\" to enumeration \"%s\"!",
              name_, type_name(tp).c_str());
 
+    # if PY_VERSION_HEX >= 0x030B0000
+    // In Python 3.11+, update the flag and bit masks by hand,
+    // since enum._proto_member.__set_name__ is not called in this code path.
+    if (t->flags & (uint32_t) enum_flags::flag_enum) {
+        int64_t flag_mask = cast<int64_t>(tp.attr("_flag_mask_"));
```

Comment:
This cast will fail when `_flag_mask_` does not fit into a signed 64 bit integer (e.g. when all bits are set). That seems to me to be a fairly common situation and should probably be covered in the tests. It might be that we need special cases for signed and unsigned integers.

Alternatively, you might be able to write this code in terms of Python `int` objects without ever casting to C++.

Reply:
More thoughts on this: Can flag enums actually be signed? If not, perhaps `uint64_t` is the right type to use.

Reply:
Can I actually cast this completely safely from overflows? I checked the Python docs, and each Py->C++ conversion (like PyLong_AsLongLong) seems to carry some overflow management.

Can I get away here by doing inplace computations like `aenum`, maybe like calling `tp.attr("_flag_mask_).attr("__iadd__")()`? I guess you meant that by avoiding the casts?

Reply:
You can cast to 64 bit signed and unsigned integers based on the signedness of the type, and that will have you covered for integers up to 64 bit (which is in any case, the most that `enum` portably supports in C++).

You will see that the existing enum code follows this pattern in some places. An alternative to this is that you can also do arithmetic directly using the wrapper types. For example, `nb::handle` et al. support `operator|`, etc.

Reply:
Currently, the in-place operators only work when the left hand side is a `nb::object ` subclass.

`tp.attr("_flag_mask_") |= val` will return a Python object with the result but cannot modify the left hand side.
This is a stupid limitation and something that could likely be fixed by adding the overloads to `nb_accessor.h`.